core/server/models/session.js

```diff
@@ -0,0 +1,54 @@
+const ghostBookshelf = require('./base');
+
+let Session,
+    Sessions;
+
+Session = ghostBookshelf.Model.extend({
+    tableName: 'sessions',
+
+    parse: function parse(attrs) {
+        if (!attrs.session_data) {
+            return attrs;
+        }
+        return Object.assign(attrs, {
```

Comment:
Curious why we use `Object.assign` here. Is there a benefit over

`
attrs.session_data = JSON.parse(attrs.session_data)
`

Just curious, let me know 🙂

Reply:
I always prefer to not mutate arguments, you never know when third party code can make assumptions e.g. in the future it's possible that bookshelf would use the difference between the input and output to calculate some properties :shrimp:

Reply:
> I always prefer to not mutate arguments

I am confused 😆But you do mutate `attrs.session_data`? `Object.assign` does just override `session_data` in the target object.

Reply:
Lol! That is suppose to be `Object.assign({}, attrs.session , { // etc...`

Reply:
I see 😁Ähhhmm i agree a bit, but this fn was designed from bookshelf to manipulate the data in+out from/to the database.